test/integration/expression/tests/zoom/interpolate-projection/test.json

```diff
@@ -1,16 +1,26 @@
 {
+  "propertySpec": {
+    "type": "projection",
+    "property-type": "data-constant",
+    "expression": {
+      "interpolated": true,
+      "parameters": [
+        "zoom"
+      ]
+    }
+  },
   "expression": [
-    "interpolate-projection",
+    "interpolate",
     [
       "linear"
     ],
     [
       "zoom"
     ],
     0,
-    "vertical-perspective",
+    ["literal", ["vertical-perspective", "vertical-perspective", 1]],
```

Comment:
This is how I made the test work - calling literal on the project array allows converting the array to a `Projection`.
I'm not sure if this is great, would love to see what you think.

Reply:
Actually, it also works with regular strings when the interpreter knows which type it should output (projection in this case).

Reply:
regular strings?

I'd like to be able to define

    10,
    "vertical-perspective"
    12, 
    "mercator"
    
Is that possible?

Reply:
Yes, yes, other tests are doing exactly that.

Reply:
I don't think there's any validation on the strings right now though... But I'm sure it won't be a big issue to add validation...

Reply:
there is no validation since I took out the .presets / .projections - it's for later.

What's special about this test, that makes it need the ["mercator", "mercator", 1] when the other tests doesn't?

Reply:
Nothing, I just wanted to see that `literal` supports `projection` casting.

Reply:
To be honest, I started with that to see that this would work and afterward saw that regular strings also works.